content/en/blog/2023/secure-apps-with-istio/index.md

```diff
@@ -0,0 +1,353 @@
+
+## What could go wrong
+
+It is critical to implement mTLS exactly as the TLS 1.3 specification dictates. Without using
+proper mTLS following the TLS specification, here are a few things that can go wrong without
+detection:
+
+### What if someone in the middle of the connection silently captures the encrypted data?
+
+If the connection doesn’t follow exactly the handshake and record protocols as outlined in TLS
+specification, for example, the connection follows the handshake protocol but not reusing the
+negotiated session key and parameters from handshake in the record protocol, you may have your
+connection’s handshake unrelated to the record protocol where identities could be different between
+the handshake and record protocols. With unrelated handshake and record protocols, one connection
+becomes two connections, thus the attack surface is increased for man-in-the middle attacks.
+
+A mTLS connection has a consistent end-to-end security from start of the handshake to finish. The
+encrypted data is encrypted with the session key negotiated using the public key in the
+certificate. Only the source and destination can decrypt the data with the private key. In other
+words, only the owner of the certificate who has the private key can decrypt the data.  Unless a
+hacker has control of the private key of the certificate, he or she doesn't have a way to mess
+around with the mTLS connection to successfully execute a man-in-the-middle attack.
+
+### What if either source or destination identity is not cryptographically secure?
+
+If the identity is based on network properties such as IP address, which could be re-allocated to
+other pods, the identity can’t be validated using cryptographic techniques. Since this type of
+identity isn’t based on cryptographic identity, your system likely has an identity cache to track
+the mapping between the identity, the pod’s network labels, the corresponding IP address and the
+Kubernetes node info where the pod is deployed. With identity cache, you could run into pod IP
+addresses being reused and identity mistaken where policy isn’t enforced properly when the identity
+cache gets out of sync for a short period of time. For example, if you don’t have cryptographic
+identity on the connection between the peers, your system would have to get the identity from the
+identity cache which could be outdated or incomplete.
+
+These Identity caches that map identity to workload IPs are not [ACID](https://en.wikipedia.org/wiki/ACID)
+(Atomicity, Consistency, Isolation, and Durability) and you want your security system to be applied
+to something with strong guarantees. Consider the following properties and questions you may want
+to ask yourself:
+
+- Staleness: How can a peer verify that an entry in the cache is 'current'?
+- Incompleteness: If there's a cache miss and the system fails to close the connection, does the
+network become unstable when it's only the cache 'synchronizer' that is failing?
```

Comment:
Do we want to emphasize **current** and **synchronizer** the same way we're emphasizing **failing closed**?
```suggestion
- Staleness: How can a peer verify that an entry in the cache is **current**?
- Incompleteness: If there's a cache miss and the system fails to close the connection, does the
network become unstable when it's only the cache **synchronizer** that is failing?
```

Reply:
Thanks!  accepted in the new commits